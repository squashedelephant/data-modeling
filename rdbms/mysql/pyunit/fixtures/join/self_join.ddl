CREATE TABLE customer(id INT NOT NULL AUTO_INCREMENT, name VARCHAR(30), business VARCHAR(20), city VARCHAR(20), state VARCHAR(2), credit_limit DECIMAL(9,2), PRIMARY KEY (id));
