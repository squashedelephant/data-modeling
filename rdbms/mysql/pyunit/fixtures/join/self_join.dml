INSERT INTO customer(name, business, city, state, credit_limit) VALUES ('Bodacious Beauty Parlor', 'beauty salon', 'Smallville', 'KS', 2000.00), ('Refreshing Cuisine', 'restaurant', 'Big City', 'MO', 5000.00);
INSERT INTO customer(name, business, city, state, credit_limit) VALUES ('Gas & Go', 'gas station', 'Smallville', 'KS', 3000.00), ('Family Pharmacy', 'pharmacy', 'Big City', 'MO', 5000.00);
INSERT INTO customer(name, business, city, state, credit_limit) VALUES ('Chill & Slurp', 'coffee shop', 'Smallville', 'KS', 1000.00), ('Hang in There', 'hospital', 'Metropolis', 'AR', 50000.00);
INSERT INTO customer(name, business, city, state, credit_limit) VALUES ('Keep the Faith', 'church', 'Big City', 'MO', 5000.00), ('Buy & Ride', 'auto dealership', 'Metropolis', 'AR', 50000.00), ('Lock & Key', 'bank', 'Big City', 'MO', 200000);
