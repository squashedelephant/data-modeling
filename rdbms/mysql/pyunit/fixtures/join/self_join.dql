SELECT DISTINCT c1.name, c1.city FROM customer AS c1 INNER JOIN customer AS c2 ON c1.city=c2.city ORDER BY c1.city ASC, c1.name ASC;
