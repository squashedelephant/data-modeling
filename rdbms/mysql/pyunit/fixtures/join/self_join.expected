Family Pharmacy
Big City
Keep the Faith
Big City
Lock & Key
Big City
Refreshing Cuisine
Big City
Buy & Ride
Metropolis
Hang in There
Metropolis
Bodacious Beauty Parlor
Smallville
Chill & Slurp
Smallville
Gas & Go
Smallville
